build-system/tasks/dist.js

```diff
@@ -128,11 +128,14 @@ async function dist() {
   await stopNailgunServer(distNailgunPort);
 
   if (argv.esm) {
-    await Promise.all([
-      createModuleCompatibleES5Bundle('v0.js'),
-      createModuleCompatibleES5Bundle('amp4ads-v0.js'),
-      createModuleCompatibleES5Bundle('shadow-v0.js'),
-    ]);
+    const bundlesToPostMutate = [createModuleCompatibleES5Bundle('v0.js')];
+    if (!argv.core_runtime_only) {
+      bundlesToPostMutate.push(
+        createModuleCompatibleES5Bundle('amp4ads-v0.js'),
+        createModuleCompatibleES5Bundle('shadow-v0.js')
+      );
+    }
+    await Promise.all(bundlesToPostMutate);
```

Comment:
Optional: These operations take O(milliseconds) time, so you could potentially simplify the code to:

```js
  if (argv.esm) {
    await createModuleCompatibleES5Bundle('v0.js');
    if (!argv.core_runtime_only) {
      await createModuleCompatibleES5Bundle('amp4ads-v0.js');
      await createModuleCompatibleES5Bundle('shadow-v0.js');
    }
  }
```